### PR TITLE
Fixing more ineffective assignments.

### PR DESCRIPTION
**Pull request by hackintoshrao**

### CODE REVIEW

File: filesystem/vfs_s3.go [hunk 1]
```diff
@@ -34,6 +34,11 @@ func (fs *S3FileSystem) Open(fl *FileLocation) (VirtualFile, error) {
 	svc := s3.New(sess)
 
 	bucketName, objectKey, err := splitS3LocationToParts(fl.Location)
+
+	if err != nil {
+		return nil, err
```
Comment (chrislusf) [inline filesystem/vfs_s3.go]: please use fmt.Errorf(...) to return a more descriptive error

File: filesystem/vfs_s3.go [hunk 2]
```diff
@@ -34,6 +34,11 @@ func (fs *S3FileSystem) Open(fl *FileLocation) (VirtualFile, error) {
 	svc := s3.New(sess)
 
 	bucketName, objectKey, err := splitS3LocationToParts(fl.Location)
+
+	if err != nil {
+		return nil, fmt.Errorf("Failed to split S3 location to parts <Location> %s: <Error> %v", fl.Location, err)
```
Comment (chrislusf) [inline filesystem/vfs_s3.go]: can you please remove <Location> and <Error>

Comment (hackintoshrao) [inline filesystem/vfs_s3.go]: > can you please remove and 

I didn't understand, can you elaborate?

Comment (chrislusf) [inline filesystem/vfs_s3.go]: return nil, fmt.Errorf("Failed to split S3 location to parts %s: %v", fl.Location, err)

**Pull request merged**